lib/makews.js

```diff
@@ -0,0 +1,49 @@
+const core = require("@actions/core");
+const exec = require("@actions/exec")
+
+async function makews(createworkspace, workspace, execfile) {
+    try {
+        let stdout = '';
+        let stderr = '';
+
+        const options = {};
+        options.listeners = {
+            stdout: (data) => {
+                stdout = data.toString();
+            },
+            stderr: (data) => {
+                stderr = data.toString();
+            }
+        };
+        options.cwd = './';
+        await exec.exec(execfile, [`workspace`, `list`], options);
+
+        if (createworkspace != "false") {
+            if (matchExact(new RegExp(`^${workspace}$`, 'gm'), stdout)) {
+                core.info(`workspace does not exists, creating as create-workspace param not set to false`)
+                await exec.exec(execfile, [`workspace`, `new`, `${workspace}`]);
+                return true
+            }
+        }
+
+        if (matchExact(new RegExp(`\\* ${workspace}$`, 'gm'), stdout)) {
+            core.info(`workspace already switched to ${workspace}`)
+            return true
+        }
+
+        core.info(`workspace already exists, switching to workspace ${workspace}`)
+        await exec.exec(execfile, [`workspace`, `select`, `${workspace}`]);
+        return true
+        
+    } catch (err) {
+        core.error(err);
+        throw err;
+    }
```

Comment:
I know that you pattern your code from the previous function. But reading now, do we really need to `catch` an and then `throwing` it again? Are we doing this this because of the `core.error()` to show a different way that it would if we just let the code fails and, consequently, the pipeline?

Reply:
yaa `core.error(err)` will fail the pipeline if this functioning error out in catch block, without this not sure if node catch will be able to pass error to core module and make pipeline fail